apps/confidential/confidential.go

```diff
@@ -799,3 +826,105 @@ func (cca Client) Account(ctx context.Context, accountID string) (Account, error
 func (cca Client) RemoveAccount(ctx context.Context, account Account) error {
 	return cca.base.RemoveAccount(ctx, account)
 }
+
+// withAdditionalCacheKeyComponents adds aditional hash for cache key
+//
+// Example:
+//
+//	params := map[string]string{
+//	    "custom_param": "custom_value",
+//	}
+//	result, err := client.AcquireTokenByCredential(ctx, scopes, confidential.withAdditionalCacheKeyComponents(params))
+func withAdditionalCacheKeyComponents(params map[string]string) interface {
+	AcquireByCredentialOption
+	AcquireSilentOption
+	options.CallOption
+} {
+
+	return struct {
+		AcquireByCredentialOption
+		AcquireSilentOption
+		options.CallOption
+	}{
+		CallOption: options.NewCallOption(
+			func(a any) error {
+				switch t := a.(type) {
+				case *acquireTokenByCredentialOptions:
+					t.cacheKeyComponents = params
+				case *acquireTokenSilentOptions:
+					t.cacheKeyComponents = params
+				default:
+					return fmt.Errorf("unexpected options type %T", a)
+				}
+				return nil
+			},
+		),
+	}
+}
+
+// WithExtraBodyParameters adds extra body parameters to the token request.
+// These parameters are added to the request body
+//
+// Example:
+//
+//	params := map[string]string{
+//	    "custom_param": "custom_value",
+//	}
+//	result, err := client.AcquireTokenByCredential(ctx, scopes, confidential.WithExtraBodyParameters(params))
+func withExtraBodyParameters(params map[string]string) interface {
```

Comment:
This and `withAdditionalCacheKeyComponents` are dead code--they're used only in tests. I suggest not adding them until you're sure you want to include them in the public API.

Reply:
Removed both the method. Initially it was planned that we will use both separate, but then now we will use `WithFMIPath` which will do things it needs.